Assets/Scripts/Buddy/BuddyStats.cs

```diff
@@ -224,7 +223,11 @@ public void AdjustHappiness( float deltaHappiness )
 
 		_happiness = Mathf.Clamp( _happiness, 0f, 1f );
 
-		actor.animator.SetFloat( "happiness", _happiness );
+		// Hidden buddy is disabled
+		if ( gameObject.activeSelf )
+		{
+			actor.animator.SetFloat( "happiness", _happiness );
+		}
```

Comment:
It spits out an error if you try to set a variable on the animator while it's disabled. I can add some logic to the reset buddy/put down so that the variables are up to date.